src/resmom/mom_walltime.c

```diff
@@ -89,22 +91,22 @@ update_walltime(job *pjob) {
     resource_def    *walltime_def;
     resource        *used_walltime;
 
-    if (0 == pjob->ji_walltime_stamp)
-        return;
-
     resources_used = &pjob->ji_wattr[(int)JOB_ATR_resc_used];
     assert(resources_used != NULL);
     walltime_def = find_resc_def(svr_resc_def, "walltime", svr_resc_size);
     assert(walltime_def != NULL);
     used_walltime = find_resc_entry(resources_used, walltime_def);
-    
+
     if (NULL == used_walltime) {
         used_walltime = add_resource_entry(resources_used, walltime_def);
         used_walltime->rs_value.at_flags |= ATR_VFLAG_SET;
         used_walltime->rs_value.at_type = ATR_TYPE_LONG;
         used_walltime->rs_value.at_val.at_long = 0;
     } else if (0 == (used_walltime->rs_value.at_flags & ATR_VFLAG_HOOK)) {
         /* walltime is not set by hook */
+        if (0 == pjob->ji_walltime_stamp)
+            return;
```

Comment:
I'm not sure I understand this.  If the walltime stamp was not set before, we leave it unset?  Why is this only for a hook?

Reply:
`walltime_stamp` is the time when we last updated walltime. A value of zero does not make sense so zero is used to represent that walltime has been stopped.

Reply:
This is the branch that hook has set walltime of the job, so we do not update it here. I reworked the conditions a little hopefully it is a bit more clear.